Display Sidewalks Mapped on Roads

thompsondt opened the pull request:
## Summary:

This pull request implements changes to accurately display sidewalk infrastructure on OpenSidewalkMap:

1. Returning all sidewalks, regardless of whether they are mapped separately or as part of the road.

2. Separating the two kinds of data into two layers, with visually distinct styling:
- All sidewalks are rendered in red.
- Sidewalks mapped on roads are indicated by a single dashed line running down the center of the road.

## Future Considerations:

1. Adding a legend to indicate the styling of sidewalks.

2. Introducing user controls to toggle the visibility of each layer.

## References:
- addresses Issue #3: Display Sidewalk Infrastructure Accurately on OpenSidewalkMap

thompsondt committed:
WIP: fix(query): broaden Overpass query to include standalone and roadway features

Changed the Overpass query to include both standalone sidewalks and sidewalks as features of roadways. Further verification needed to ensure it meets acceptance criteria.

Refs: #3

thompsondt committed:
Merge branch 'main' into fix-sidwalks-not-displaying-on-tagged-roads

thompsondt committed:
fix: show sidewalks on roads

- sidewalks tagged on roads (rather than as separate paths) will now be included as dashed lines

CurtisGreen commented on src/components/MainMap.tsx:
```diff
@@ -98,14 +99,38 @@ export const MainMap = ({
+              "line-color": "red",
+              "line-opacity": 0.8,
+              "line-width": 2,
+              "line-dasharray": [1, 1 ],
```
Nit
```suggestion
              "line-dasharray": [1, 1],
```

thompsondt replied:
good spot there. I'll look for a config for the linter that spots this as well.

CurtisGreen commented on src/components/MainMap.tsx:
```diff
@@ -98,14 +99,38 @@ export const MainMap = ({
-            id="Data"
+            id="Paths"
             type="line"
+            filter={[
+              "match",
+                ["get", "highway"],
+                paths,
+                true,
+                false
+              ]} // Filter for true paths
```
What do the positional arguments of true / false do here? I found documentation for filter & match but I wasn't able to figure out what those do

thompsondt replied:
It took me a while to figure out the filters. It looks like it's possible to specify the return value independently for each match. In our case, I'm returning true for ANY match in the paths array. It could be more appropriately formatted like this. https://docs.mapbox.com/style-spec/reference/expressions/#match

```
filter={[
              "match",
                ["get", "highway"],
                paths, true, // return true if match found in the array of recognized path values
                false // return false as a fallback
              ]} // Filter for true paths
```

CurtisGreen reviewed (commented):
Very cool, I already found a small road that had both on-road and separate sidewalk markings so I can go fix that

thompsondt committed:
style: groom mapbox filter settings

- adds a little explanation about how the match filter works
- removes trailing whitespace after the last item in an array

thompsondt commented:
I looked into activating some additional rules with ESLint, but that might be done more cleanly under a separate branch and PR.
